builtin/providers/cloudstack/resource_cloudstack_loadbalancer_test.go

```diff
@@ -44,6 +45,14 @@ func TestAccCloudStackLoadBalancerRule_update(t *testing.T) {
 				Config: testAccCloudStackLoadBalancerRule_basic,
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckCloudStackLoadBalancerRuleExist("cloudstack_loadbalancer_rule.foo"),
+					func(s *terraform.State) error {
+						rs, ok := s.RootModule().Resources["cloudstack_loadbalancer_rule.foo"]
+						if !ok {
+							return fmt.Errorf("Not found: cloudstack_loadbalancer_rule.foo")
+						}
+						id = rs.Primary.ID
+						return nil
+					},
```

Comment:
Can you elaborate on this one? I would say that the previous check `testAccCheckCloudStackLoadBalancerRuleExist("cloudstack_loadbalancer_rule.foo")` should already cover this check?

And if not it's probably better to improve the `testAccCheckCloudStackLoadBalancerRuleExist` function instead of adding this custom inline one...